shapash/explainer/smart_plotter.py

```diff
@@ -2323,3 +2325,211 @@ def correlations(
+            raise AssertionError("Multi-class classification is not supported")
+
+        self.explainer.compute_features_stability(selection)
+
+        dataset = self.explainer.x_init
+
+        ordinal = lambda n: "%d%s" % (n, "tsnrhtdd"[(math.floor(n / 10) % 10 != 1) * (n % 10 < 4) * n % 10:: 4])
+
+        # Check if entry is a single instance or not
+        if len(selection) == 1:
+            # Compute explanations for instance and neighbors
+            g = self.explainer.features_stability["norm_shap"]
+
+            # Reorder indices based on absolute values of the 1st row (i.e. the instance) in descending order
+            inds = np.flip(np.abs(g[0, :]).argsort())
+            g = g[:, inds]
+            columns = [dataset.columns[i] for i in inds]
+
+            # Plot
+            g_df = pd.DataFrame(g, columns=columns).T.rename(
+                    columns={
+                        **{0: "instance", 1: "closest neighbor"},
+                        **{i: ordinal(i) + " closest neighbor" for i in range(2, len(g))},
+                    }
+                )
+
+            # Keep only max_features
+            if max_features is not None: g_df = g_df[:max_features]
+
+            fig = go.Figure(data=[go.Bar(name=g_df.iloc[::-1, ::-1].columns[i],
+                            y=g_df.iloc[::-1, ::-1].index.tolist(),
+                            x=g_df.iloc[::-1, ::-1].iloc[:, i],
+                            orientation='h') for i in range(g_df.shape[1])])
+
+            fig.update_layout(title={'text': "Explanation of local stability:\
+                 How similar are explanations for closeby neighbours?"},
+                              xaxis_title="Normalized SHAP value",
+                              barmode="group",
+                              bargap=0.5,
+                              height=30*g_df.shape[0]*g_df.shape[1] if g_df.shape[0] < 100
+                              else 10*g_df.shape[0]*g_df.shape[1],
+                              width=900,
+                              legend={"traceorder": "reversed"},
+                              xaxis={"side": "top"},
+                              margin=dict(t=185))
+
+            fig.show()
+
+        else:
+            variability = self.explainer.features_stability["variability"]
+            amplitude = self.explainer.features_stability["amplitude"]
+
+            mean_variability = variability.mean(axis=0)
+            mean_amplitude = amplitude.mean(axis=0)
+
+            # If set, only keep features with the highest mean amplitude
+            if max_features is not None:
+                keep = mean_amplitude.argsort()[::-1][:max_features]
+                keep = np.sort(keep)
+
+                variability = variability[:, keep]
+                mean_variability = mean_variability[keep]
+                amplitude = amplitude[:, keep]
+                mean_amplitude = mean_amplitude[keep]
+                dataset = dataset.iloc[:, keep]
+
```

Comment:
To have clearer plots, use the label of each feature (self.explainer.features_dict attribute)

Reply:
can you change the colors, colors scale, style to apply the Shapash graphic charter?